From: Alan Curry <pacman@theworld.com>
Date: Wed, 25 Jul 2007 01:28:32 +0000 (+1000)
Subject: [POWERPC] Fix Pegasos keyboard detection
X-Git-Url: http://git.kernel.org/?p=linux%2Fkernel%2Fgit%2Ftorvalds%2Flinux-2.6.git;a=commitdiff_plain;h=f5d834fc34e61f1a40435981062000e5d2b2baa8

[POWERPC] Fix Pegasos keyboard detection

As of 2.6.22 the kernel doesn't recognize the i8042 keyboard/mouse
controller on the PegasosPPC.  This is because of a feature/bug in the
OF device tree: the "device_type" attribute is an empty string instead
of "8042" as the kernel expects.  This adds a secondary detection
which looks for a device whose *name* is "8042" if there is no device
whose *type* is "8042".

Signed-off-by: Alan Curry <pacman@world.std.com>
Acked-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
Signed-off-by: Paul Mackerras <paulus@samba.org>
---

[ adapted to 2.6.22 -- tbm ]

--- a/arch/powerpc/kernel/setup-common.c	2007-07-31 13:19:04.000000000 +0200
+++ b/arch/powerpc/kernel/setup-common.c	2007-07-31 13:19:36.000000000 +0200
@@ -487,6 +487,10 @@
 	switch(base_port) {
 	case I8042_DATA_REG:
 		np = of_find_node_by_type(NULL, "8042");
+		/* Pegasos has no device_type on its 8042 node, look for the
+		 * name instead */
+		if (!np)
+			np = of_find_node_by_name(NULL, "8042");
 		break;
 	case FDC_BASE: /* FDC1 */
 		np = of_find_node_by_type(NULL, "fdc");

